src/Microsoft.Framework.TypeNameHelper.Internal/TypeNameHelper.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.Framework.Internal
+{
+    internal class TypeNameHelper
+    {
+        private static readonly Dictionary<Type, string> _builtInTypeNames = new Dictionary<Type, string>
+            {
+            { typeof(bool), "bool" },
+            { typeof(byte), "byte" },
+            { typeof(char), "char" },
+            { typeof(decimal), "decimal" },
+            { typeof(double), "double" },
+            { typeof(float), "float" },
+            { typeof(int), "int" },
+            { typeof(long), "long" },
+            { typeof(object), "object" },
+            { typeof(sbyte), "sbyte" },
+            { typeof(short), "short" },
+            { typeof(string), "string" },
+            { typeof(uint), "uint" },
+            { typeof(ulong), "ulong" },
+            { typeof(ushort), "ushort" }
+            };
+
+        public static string GetTypeDisplayName(Type type, bool fullname = true)
+        {
+            var sb = new StringBuilder();
+            ProcessTypeName(type, sb, fullname);
+            return sb.ToString();
+        }
+
+        private static void AppendGenericArguments(Type[] args, int startIndex, int nArgsToAppend, StringBuilder sb, bool fullName)
+        {
+            var nArgs = args.Length;
```

Comment:
This is already merged. Should I make corrections and submit new pull req for it?